src/Silex/Application.php

```diff
@@ -495,9 +495,11 @@ public function mount($prefix, $controllers)
     {
         if ($controllers instanceof ControllerProviderInterface) {
             $controllers = $controllers->connect($this);
-        }
 
-        if (!$controllers instanceof ControllerCollection) {
+            if (!$controllers instanceof ControllerCollection) {
+                throw new \LogicException('The ControllerProviderInterface must return a ControllerCollection instance for the "connect" method.');
```

Comment:
It would be even more explicit if we could get the actual class name.

```
The method "MyControllerProvider::connect" must return a "ControllerCollection"
```